File: pubnub-java/src/main/java/com/pubnub/api/endpoints/objects_api/channel/RemoveChannelMetadata.java
package com.pubnub.api.endpoints.objects_api.channel;

import com.pubnub.api.endpoints.BuilderSteps;
import com.pubnub.api.endpoints.Endpoint;
import com.pubnub.api.endpoints.remoteaction.ExtendedRemoteAction;
import com.pubnub.api.endpoints.remoteaction.MappingRemoteAction;
import com.pubnub.api.models.consumer.objects_api.channel.PNRemoveChannelMetadataResult;
import com.pubnub.internal.InternalPubNubClient;
import lombok.AllArgsConstructor;

public class RemoveChannelMetadata extends Endpoint<PNRemoveChannelMetadataResult> {
    public RemoveChannelMetadata(String channel, final InternalPubNubClient pubnubInstance) {
        super(pubnubInstance);
        this.channel = channel;
    }

    private final String channel;

    @Override
    protected ExtendedRemoteAction<PNRemoveChannelMetadataResult> createAction() {
        return new MappingRemoteAction<>(
                pubnub.removeChannelMetadata(channel),
                result -> new PNRemoveChannelMetadataResult(
                        result.getStatus(),
                        null
                )
        );
    }

    public static Builder builder(final InternalPubNubClient pubnubInstance) {
        return new Builder(pubnubInstance);
    }

    @AllArgsConstructor
    public static class Builder implements BuilderSteps.ChannelStep<RemoveChannelMetadata> {
        private final InternalPubNubClient pubnubInstance;

        @Override
        public RemoveChannelMetadata channel(final String channel) {
            return new RemoveChannelMetadata(channel, pubnubInstance);
        }
    }
}

File: pubnub-kotlin/src/main/kotlin/com/pubnub/api/endpoints/objects/channel/RemoveChannelMetadata.kt
package com.pubnub.api.endpoints.objects.channel

import com.pubnub.api.Endpoint
import com.pubnub.api.endpoints.remoteaction.map
import com.pubnub.api.models.consumer.objects.PNRemoveMetadataResult
import com.pubnub.internal.DelegatingEndpoint
import com.pubnub.internal.endpoints.objects.channel.IRemoveChannelMetadata
import com.pubnub.internal.endpoints.objects.channel.RemoveChannelMetadata

class RemoveChannelMetadata internal constructor(private val removeChannelMetadata: RemoveChannelMetadata) :
    DelegatingEndpoint<PNRemoveMetadataResult>(), IRemoveChannelMetadata by removeChannelMetadata {
    override fun createAction(): Endpoint<PNRemoveMetadataResult> = removeChannelMetadata.map(PNRemoveMetadataResult::from)
}